src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -561,6 +561,8 @@ public function getEditLink($id)
             $id
         );
 
+        $link = $this->gridField->addRequestToURL($link);
```

Comment:
Initially, I implemented this behaviour in `GridFieldStateManager` class in `addStateToURL` method, but then I've decided to move it to the GridField, because it just collects parameters from request to add to link. Then this link will be passed to addStateToURL() to add new hash to access new state for new GridField.